### DISCUSSION TITLE
IKEv2 Virtual ip no outgoing encrypted packets. No communications

### DISCUSSION THREAD

**Discussion by Viktort-rf** [strongswan/strongswan#1339] [Q&A] [answered]
I set up ikev2 virtual ip push on server in scenario RW. But I do not see outgoing encrypted packet in "swanctl -l", so haven't communications between GW and RW. Any traffic direction (forward, input)

Distr - Ubuntu 20.04.4 LTS
Kernel - 5.4.0-104-generic
ip - iproute2-ss200127
strongSwan swanctl 5.8.2

swanctl.conf
```
swanctl.conf:
        ipsec {
                version = 2
                local_addrs = "GW-public-ip"
                remote_addrs = %any
                mobike = yes
                fragmentation = yes
                keyingtries = 3
                unique = keep
                proposals = aes256-aes192-aes128-sha512-sha384-sha256-modp2048-modp3072-modp4096-modp6144-modp8192
                pools = ipsec-ikev2
                dpd_delay = 10s
                send_cert = always
                local-psk {
                        auth = pubkey
                        certs = vpn-cert-full.pem
                        id = vpn.example.com
                }
                remote-eap {
                        eap_id = %any
                        auth = eap-mschapv2
                }
                children {
                        ipsec {
                                start_action = start
                                mode = tunnel
                                esp_proposals = aes256-aes192-aes128-sha512-sha384-sha256-modp2048-modp3072-modp4096-modp6144-modp8192
                                local_ts = 0.0.0.0/0
                                remote_ts = dynamic
                                mark_in = %unique
                                mark_out = %unique
                                dpd_action = clear
                        }

                }
        }
}
pools {
        ipsec-ikev2 {
                addrs = 192.168.255.128/25
                dns = 192.168.254.1
        }
}
authorities {
}
secrets {
        eap-testuser {
                id = username-test
                secret = strongpass
        }
}

```

swanctl -l 

> ipsec: #33, ESTABLISHED, IKEv2, 72d8c090224a9d42_i e9c0c686975eaacb_r*
>   local  'vpn.example.com' @ "GW-public-ip"[4500]
>   remote '192.168.23.43' @ "RW_CG-NAT_Public-ip"[27945] EAP: 'username-test' [192.168.255.129]
>   AES_CBC-256/HMAC_SHA2_256_128/PRF_HMAC_SHA2_256/MODP_2048
>   established 14s ago, rekeying in 13432s
>   ipsec: #55, reqid 18, INSTALLED, TUNNEL-in-UDP, ESP:AES_CBC-256/HMAC_SHA2_256_128
>     installed 14s ago, rekeying in 3387s, expires in 3946s
>     **in  c3a72ec6 (0x00000012),   1816 bytes,    26 packets**
>     _**out 0e399575 (0x00000012),      0 bytes,     0 packets**_
>     local  0.0.0.0/0
>     remote 192.168.255.129/32

ip -s xfrm policy

```
src 0.0.0.0/0 dst 192.168.255.129/32 uid 0
        dir out action allow index 7169 priority 383615 share any flag  (0x00000000)
        lifetime config:
          limit: soft (INF)(bytes), hard (INF)(bytes)
          limit: soft (INF)(packets), hard (INF)(packets)
          expire add: soft 0(sec), hard 0(sec)
          expire use: soft 0(sec), hard 0(sec)
        lifetime current:
          0(bytes), 0(packets)
          add 2022-10-14 10:27:33 use -
        mark 0x12/0xffffffff 
        tmpl src "GW-public-ip" dst "RW_CG-NAT_Public-ip"
                proto esp spi 0x0e399575(238654837) reqid 18(0x00000012) mode tunnel
                level required share any 
                enc-mask ffffffff auth-mask ffffffff comp-mask ffffffff
src 192.168.255.129/32 dst 0.0.0.0/0 uid 0
        dir fwd action allow index 7162 priority 383615 share any flag  (0x00000000)
        lifetime config:
          limit: soft (INF)(bytes), hard (INF)(bytes)
          limit: soft (INF)(packets), hard (INF)(packets)
          expire add: soft 0(sec), hard 0(sec)
          expire use: soft 0(sec), hard 0(sec)
        lifetime current:
          0(bytes), 0(packets)
          add 2022-10-14 10:27:33 use -
        mark 0x12/0xffffffff 
        tmpl src "RW_CG-NAT_Public-ip" dst "GW-public-ip"
                proto esp spi 0x00000000(0) reqid 18(0x00000012) mode tunnel
                level required share any 
                enc-mask ffffffff auth-mask ffffffff comp-mask ffffffff
src 192.168.255.129/32 dst 0.0.0.0/0 uid 0
        dir in action allow index 7152 priority 383615 share any flag  (0x00000000)
        lifetime config:
          limit: soft (INF)(bytes), hard (INF)(bytes)
          limit: soft (INF)(packets), hard (INF)(packets)
          expire add: soft 0(sec), hard 0(sec)
          expire use: soft 0(sec), hard 0(sec)
        lifetime current:
          0(bytes), 0(packets)
          add 2022-10-14 10:27:33 use -
        mark 0x12/0xffffffff 
        tmpl src "RW_CG-NAT_Public-ip" dst "GW-public-ip"
                proto esp spi 0x00000000(0) reqid 18(0x00000012) mode tunnel
                level required share any 
                enc-mask ffffffff auth-mask ffffffff comp-mask ffffffff
```

ip route show table 220

> 192.168.255.129 via "GW_FOR_SERVERS" dev uplink-1 proto static

iptables-save

> # Generated by iptables-save v1.8.4 on Fri Oct 14 11:04:15 2022
> *mangle
> :PREROUTING ACCEPT [10162:1681860]
> :INPUT ACCEPT [8688:1192944]
> :FORWARD ACCEPT [1403:484363]
> :OUTPUT ACCEPT [1944:582439]
> :POSTROUTING ACCEPT [3347:1066802]
> -A PREROUTING -s "Other-public-ip-working-l2tp-client" -d "GW-public-ip" -p udp -m udp --sport 23424 --dport 4500 -j MARK --set-xmark 0x67/0xffffffff
> -A INPUT -s "Other-public-ip-working-l2tp-client" -d "GW-public-ip" -m policy --dir in --pol ipsec --spi 0x683dbece -j CONNMARK --set-xmark 0x67/0xffffffff
> -A OUTPUT -s "GW-public-ip" -d "Other-public-ip-working-l2tp-client" -m mark --mark 0x0 -j CONNMARK --restore-mark --nfmask 0xffffffff --ctmask 0xffffffff
> COMMIT
> # Completed on Fri Oct 14 11:04:15 2022
> # Generated by iptables-save v1.8.4 on Fri Oct 14 11:04:15 2022
> *nat
> :PREROUTING ACCEPT [75476:4650159]
> :INPUT ACCEPT [217:66336]
> :OUTPUT ACCEPT [229:56832]
> :POSTROUTING ACCEPT [3278:277691]
> :DOCKER - [0:0]
> -A PREROUTING -m addrtype --dst-type LOCAL -j DOCKER
> -A OUTPUT ! -d 127.0.0.0/8 -m addrtype --dst-type LOCAL -j DOCKER
> -A POSTROUTING -s 172.21.0.0/16 ! -o br-2e64e140f6d0 -j MASQUERADE
> -A POSTROUTING -s 172.20.0.0/16 ! -o br-d5387274a0d9 -j MASQUERADE
> -A POSTROUTING -s 172.17.0.0/16 ! -o docker0 -j MASQUERADE
> -A POSTROUTING -s 172.19.0.0/16 ! -o br-e2c86a01077a -j MASQUERADE
> -A POSTROUTING -s 172.18.0.0/16 ! -o br-24a66e558eb0 -j MASQUERADE
> -A POSTROUTING -s 192.168.255.0/24 -o uplink-1 -j MASQUERADE
> -A POSTROUTING -s 192.168.254.0/24 -o uplink-1 -j MASQUERADE
> -A POSTROUTING -s 192.168.23.0/24 -o uplink-1 -j MASQUERADE
> -A POSTROUTING -s 172.18.0.2/32 -d 172.18.0.2/32 -p tcp -m tcp --dport 9000 -j MASQUERADE
> -A POSTROUTING -s 172.20.0.2/32 -d 172.20.0.2/32 -p tcp -m tcp --dport 9116 -j MASQUERADE
> -A POSTROUTING -s 172.20.0.3/32 -d 172.20.0.3/32 -p tcp -m tcp --dport 9093 -j MASQUERADE
> -A POSTROUTING -s 172.20.0.5/32 -d 172.20.0.5/32 -p tcp -m tcp --dport 3000 -j MASQUERADE
> -A POSTROUTING -s 172.20.0.4/32 -d 172.20.0.4/32 -p tcp -m tcp --dport 9090 -j MASQUERADE
> -A POSTROUTING -s 172.21.0.2/32 -d 172.21.0.2/32 -p tcp -m tcp --dport 443 -j MASQUERADE
> -A POSTROUTING -s 172.21.0.2/32 -d 172.21.0.2/32 -p tcp -m tcp --dport 3000 -j MASQUERADE
> -A POSTROUTING -s 172.21.0.2/32 -d 172.21.0.2/32 -p udp -m udp --dport 53 -j MASQUERADE
> -A POSTROUTING -s 172.21.0.2/32 -d 172.21.0.2/32 -p tcp -m tcp --dport 80 -j MASQUERADE
> -A POSTROUTING -s 172.21.0.2/32 -d 172.21.0.2/32 -p tcp -m tcp --dport 53 -j MASQUERADE
> -A DOCKER -i br-2e64e140f6d0 -j RETURN
> -A DOCKER -i br-d5387274a0d9 -j RETURN
> -A DOCKER -i docker0 -j RETURN
> -A DOCKER -i br-e2c86a01077a -j RETURN
> -A DOCKER -i br-24a66e558eb0 -j RETURN
> -A DOCKER -d 192.168.254.1/32 ! -i br-24a66e558eb0 -p tcp -m tcp --dport 9000 -j DNAT --to-destination 172.18.0.2:9000
> -A DOCKER -d 192.168.254.1/32 ! -i br-d5387274a0d9 -p tcp -m tcp --dport 9116 -j DNAT --to-destination 172.20.0.2:9116
> -A DOCKER -d 192.168.254.1/32 ! -i br-d5387274a0d9 -p tcp -m tcp --dport 9093 -j DNAT --to-destination 172.20.0.3:9093
> -A DOCKER -d 192.168.254.1/32 ! -i br-d5387274a0d9 -p tcp -m tcp --dport 3001 -j DNAT --to-destination 172.20.0.5:3000
> -A DOCKER -d 192.168.254.1/32 ! -i br-d5387274a0d9 -p tcp -m tcp --dport 9090 -j DNAT --to-destination 172.20.0.4:9090
> -A DOCKER -d 192.168.254.1/32 ! -i br-2e64e140f6d0 -p tcp -m tcp --dport 4433 -j DNAT --to-destination 172.21.0.2:443
> -A DOCKER -d 192.168.254.1/32 ! -i br-2e64e140f6d0 -p tcp -m tcp --dport 3000 -j DNAT --to-destination 172.21.0.2:3000
> -A DOCKER -d 192.168.254.1/32 ! -i br-2e64e140f6d0 -p udp -m udp --dport 53 -j DNAT --to-destination 172.21.0.2:53
> -A DOCKER -d 192.168.254.1/32 ! -i br-2e64e140f6d0 -p tcp -m tcp --dport 8080 -j DNAT --to-destination 172.21.0.2:80
> -A DOCKER -d 192.168.254.1/32 ! -i br-2e64e140f6d0 -p tcp -m tcp --dport 53 -j DNAT --to-destination 172.21.0.2:53
> COMMIT
> # Completed on Fri Oct 14 11:04:15 2022
> # Generated by iptables-save v1.8.4 on Fri Oct 14 11:04:15 2022
> *filter
> :INPUT DROP [22661:1354935]
> :FORWARD ACCEPT [0:0]
> :OUTPUT ACCEPT [0:0]
> :DOCKER - [0:0]
> :DOCKER-ISOLATION-STAGE-1 - [0:0]
> :DOCKER-ISOLATION-STAGE-2 - [0:0]
> :DOCKER-USER - [0:0]
> :ufw-after-forward - [0:0]
> :ufw-after-input - [0:0]
> :ufw-after-logging-forward - [0:0]
> :ufw-after-logging-input - [0:0]
> :ufw-after-logging-output - [0:0]
> :ufw-after-output - [0:0]
> :ufw-before-forward - [0:0]
> :ufw-before-input - [0:0]
> :ufw-before-logging-forward - [0:0]
> :ufw-before-logging-input - [0:0]
> :ufw-before-logging-output - [0:0]
> :ufw-before-output - [0:0]
> :ufw-logging-allow - [0:0]
> :ufw-logging-deny - [0:0]
> :ufw-not-local - [0:0]
> :ufw-reject-forward - [0:0]
> :ufw-reject-input - [0:0]
> :ufw-reject-output - [0:0]
> :ufw-skip-to-policy-forward - [0:0]
> :ufw-skip-to-policy-input - [0:0]
> :ufw-skip-to-policy-output - [0:0]
> :ufw-track-forward - [0:0]
> :ufw-track-input - [0:0]
> :ufw-track-output - [0:0]
> :ufw-user-forward - [0:0]
> :ufw-user-input - [0:0]
> :ufw-user-limit - [0:0]
> :ufw-user-limit-accept - [0:0]
> :ufw-user-logging-forward - [0:0]
> :ufw-user-logging-input - [0:0]
> :ufw-user-logging-output - [0:0]
> :ufw-user-output - [0:0]
> -A INPUT -j ufw-before-logging-input
> -A INPUT -j ufw-before-input
> -A INPUT -j ufw-after-input
> -A INPUT -j ufw-after-logging-input
> -A INPUT -j ufw-reject-input
> -A INPUT -j ufw-track-input
> -A FORWARD -j DOCKER-USER
> -A FORWARD -j DOCKER-ISOLATION-STAGE-1
> -A FORWARD -o br-2e64e140f6d0 -m conntrack --ctstate RELATED,ESTABLISHED -j ACCEPT
> -A FORWARD -o br-2e64e140f6d0 -j DOCKER
> -A FORWARD -i br-2e64e140f6d0 ! -o br-2e64e140f6d0 -j ACCEPT
> -A FORWARD -i br-2e64e140f6d0 -o br-2e64e140f6d0 -j ACCEPT
> -A FORWARD -o br-d5387274a0d9 -m conntrack --ctstate RELATED,ESTABLISHED -j ACCEPT
> -A FORWARD -o br-d5387274a0d9 -j DOCKER
> -A FORWARD -i br-d5387274a0d9 ! -o br-d5387274a0d9 -j ACCEPT
> -A FORWARD -i br-d5387274a0d9 -o br-d5387274a0d9 -j ACCEPT
> -A FORWARD -o docker0 -m conntrack --ctstate RELATED,ESTABLISHED -j ACCEPT
> -A FORWARD -o docker0 -j DOCKER
> -A FORWARD -i docker0 ! -o docker0 -j ACCEPT
> -A FORWARD -i docker0 -o docker0 -j ACCEPT
> -A FORWARD -o br-e2c86a01077a -m conntrack --ctstate RELATED,ESTABLISHED -j ACCEPT
> -A FORWARD -o br-e2c86a01077a -j DOCKER
> -A FORWARD -i br-e2c86a01077a ! -o br-e2c86a01077a -j ACCEPT
> -A FORWARD -i br-e2c86a01077a -o br-e2c86a01077a -j ACCEPT
> -A FORWARD -o br-24a66e558eb0 -m conntrack --ctstate RELATED,ESTABLISHED -j ACCEPT
> -A FORWARD -o br-24a66e558eb0 -j DOCKER
> -A FORWARD -i br-24a66e558eb0 ! -o br-24a66e558eb0 -j ACCEPT
> -A FORWARD -i br-24a66e558eb0 -o br-24a66e558eb0 -j ACCEPT
> -A FORWARD -j ufw-before-logging-forward
> -A FORWARD -j ufw-before-forward
> -A FORWARD -j ufw-after-forward
> -A FORWARD -j ufw-after-logging-forward
> -A FORWARD -j ufw-reject-forward
> -A FORWARD -j ufw-track-forward
> -A OUTPUT -j ufw-before-logging-output
> -A OUTPUT -j ufw-before-output
> -A OUTPUT -j ufw-after-output
> -A OUTPUT -j ufw-after-logging-output
> -A OUTPUT -j ufw-reject-output
> -A OUTPUT -j ufw-track-output
> -A DOCKER -d 172.18.0.2/32 ! -i br-24a66e558eb0 -o br-24a66e558eb0 -p tcp -m tcp --dport 9000 -j ACCEPT
> -A DOCKER -d 172.20.0.2/32 ! -i br-d5387274a0d9 -o br-d5387274a0d9 -p tcp -m tcp --dport 9116 -j ACCEPT
> -A DOCKER -d 172.20.0.3/32 ! -i br-d5387274a0d9 -o br-d5387274a0d9 -p tcp -m tcp --dport 9093 -j ACCEPT
> -A DOCKER -d 172.20.0.5/32 ! -i br-d5387274a0d9 -o br-d5387274a0d9 -p tcp -m tcp --dport 3000 -j ACCEPT
> -A DOCKER -d 172.20.0.4/32 ! -i br-d5387274a0d9 -o br-d5387274a0d9 -p tcp -m tcp --dport 9090 -j ACCEPT
> -A DOCKER -d 172.21.0.2/32 ! -i br-2e64e140f6d0 -o br-2e64e140f6d0 -p tcp -m tcp --dport 443 -j ACCEPT
> -A DOCKER -d 172.21.0.2/32 ! -i br-2e64e140f6d0 -o br-2e64e140f6d0 -p tcp -m tcp --dport 3000 -j ACCEPT
> -A DOCKER -d 172.21.0.2/32 ! -i br-2e64e140f6d0 -o br-2e64e140f6d0 -p udp -m udp --dport 53 -j ACCEPT
> -A DOCKER -d 172.21.0.2/32 ! -i br-2e64e140f6d0 -o br-2e64e140f6d0 -p tcp -m tcp --dport 80 -j ACCEPT
> -A DOCKER -d 172.21.0.2/32 ! -i br-2e64e140f6d0 -o br-2e64e140f6d0 -p tcp -m tcp --dport 53 -j ACCEPT
> -A DOCKER-ISOLATION-STAGE-1 -i br-2e64e140f6d0 ! -o br-2e64e140f6d0 -j DOCKER-ISOLATION-STAGE-2
> -A DOCKER-ISOLATION-STAGE-1 -i br-d5387274a0d9 ! -o br-d5387274a0d9 -j DOCKER-ISOLATION-STAGE-2
> -A DOCKER-ISOLATION-STAGE-1 -i docker0 ! -o docker0 -j DOCKER-ISOLATION-STAGE-2
> -A DOCKER-ISOLATION-STAGE-1 -i br-e2c86a01077a ! -o br-e2c86a01077a -j DOCKER-ISOLATION-STAGE-2
> -A DOCKER-ISOLATION-STAGE-1 -i br-24a66e558eb0 ! -o br-24a66e558eb0 -j DOCKER-ISOLATION-STAGE-2
> -A DOCKER-ISOLATION-STAGE-1 -j RETURN
> -A DOCKER-ISOLATION-STAGE-2 -o br-2e64e140f6d0 -j DROP
> -A DOCKER-ISOLATION-STAGE-2 -o br-d5387274a0d9 -j DROP
> -A DOCKER-ISOLATION-STAGE-2 -o docker0 -j DROP
> -A DOCKER-ISOLATION-STAGE-2 -o br-e2c86a01077a -j DROP
> -A DOCKER-ISOLATION-STAGE-2 -o br-24a66e558eb0 -j DROP
> -A DOCKER-ISOLATION-STAGE-2 -j RETURN
> -A DOCKER-USER -j RETURN
> -A ufw-after-input -p udp -m udp --dport 137 -j ufw-skip-to-policy-input
> -A ufw-after-input -p udp -m udp --dport 138 -j ufw-skip-to-policy-input
> -A ufw-after-input -p tcp -m tcp --dport 139 -j ufw-skip-to-policy-input
> -A ufw-after-input -p tcp -m tcp --dport 445 -j ufw-skip-to-policy-input
> -A ufw-after-input -p udp -m udp --dport 67 -j ufw-skip-to-policy-input
> -A ufw-after-input -p udp -m udp --dport 68 -j ufw-skip-to-policy-input
> -A ufw-after-input -m addrtype --dst-type BROADCAST -j ufw-skip-to-policy-input
> -A ufw-before-forward -m conntrack --ctstate RELATED,ESTABLISHED -j ACCEPT
> -A ufw-before-forward -p icmp -m icmp --icmp-type 3 -j ACCEPT
> -A ufw-before-forward -p icmp -m icmp --icmp-type 11 -j ACCEPT
> -A ufw-before-forward -p icmp -m icmp --icmp-type 12 -j ACCEPT
> -A ufw-before-forward -p icmp -m icmp --icmp-type 8 -j ACCEPT
> -A ufw-before-forward -j ufw-user-forward
> -A ufw-before-input -i lo -j ACCEPT
> -A ufw-before-input -m conntrack --ctstate RELATED,ESTABLISHED -j ACCEPT
> -A ufw-before-input -m conntrack --ctstate INVALID -j ufw-logging-deny
> -A ufw-before-input -m conntrack --ctstate INVALID -j DROP
> -A ufw-before-input -p icmp -m icmp --icmp-type 3 -j ACCEPT
> -A ufw-before-input -p icmp -m icmp --icmp-type 11 -j ACCEPT
> -A ufw-before-input -p icmp -m icmp --icmp-type 12 -j ACCEPT
> -A ufw-before-input -p icmp -m icmp --icmp-type 8 -j ACCEPT
> -A ufw-before-input -d "GW-public-ip" -p udp -m udp --dport 1701 -m policy --dir in --pol ipsec -j ACCEPT
> -A ufw-before-input -d "GW-public-ip" -p udp -m udp --dport 1701 -j DROP
> -A ufw-before-input -p udp -m udp --sport 67 --dport 68 -j ACCEPT
> -A ufw-before-input -j ufw-not-local
> -A ufw-before-input -d 224.0.0.251/32 -p udp -m udp --dport 5353 -j ACCEPT
> -A ufw-before-input -d 239.255.255.250/32 -p udp -m udp --dport 1900 -j ACCEPT
> -A ufw-before-input -j ufw-user-input
> -A ufw-before-output -o lo -j ACCEPT
> -A ufw-before-output -m conntrack --ctstate RELATED,ESTABLISHED -j ACCEPT
> -A ufw-before-output -j ufw-user-output
> -A ufw-not-local -m addrtype --dst-type LOCAL -j RETURN
> -A ufw-not-local -m addrtype --dst-type MULTICAST -j RETURN
> -A ufw-not-local -m addrtype --dst-type BROADCAST -j RETURN
> -A ufw-not-local -m limit --limit 3/min --limit-burst 10 -j ufw-logging-deny
> -A ufw-not-local -j DROP
> -A ufw-skip-to-policy-forward -j ACCEPT
> -A ufw-skip-to-policy-input -j DROP
> -A ufw-skip-to-policy-output -j ACCEPT
> -A ufw-track-forward -p tcp -m conntrack --ctstate NEW -j ACCEPT
> -A ufw-track-forward -p udp -m conntrack --ctstate NEW -j ACCEPT
> -A ufw-track-output -p tcp -m conntrack --ctstate NEW -j ACCEPT
> -A ufw-track-output -p udp -m conntrack --ctstate NEW -j ACCEPT
> -A ufw-user-input -p tcp -m tcp --dport 48657 -j ACCEPT
> -A ufw-user-input -p udp -m udp --dport 4500 -j ACCEPT
> -A ufw-user-input -p udp -m udp --dport 500 -j ACCEPT
> -A ufw-user-input -d 192.168.254.1/32 -p tcp -m tcp --dport 53 -j ACCEPT
> -A ufw-user-input -d 192.168.254.1/32 -p udp -m udp --dport 53 -j ACCEPT
> -A ufw-user-input -s 192.168.23.0/24 -j ACCEPT
> -A ufw-user-input -s 192.168.255.0/30 -j ACCEPT
> -A ufw-user-input -d "GW-public-ip" -p tcp -m tcp --dport 7335 -j ACCEPT
> -A ufw-user-input -d "GW-public-ip" -p tcp -m tcp --dport 13165 -j ACCEPT
> -A ufw-user-input -d "GW-public-ip" -p udp -m udp --dport 13165 -j ACCEPT
> -A ufw-user-input -s 172.16.0.0/12 -d 192.168.254.1/32 -p tcp -m tcp --dport 9100 -j ACCEPT
> -A ufw-user-input -s 172.16.0.0/12 -d 192.168.254.1/32 -p udp -m udp --dport 9100 -j ACCEPT
> -A ufw-user-input -p tcp -m tcp --dport 443 -j ACCEPT
> -A ufw-user-input -p udp -m udp --dport 443 -j ACCEPT
> -A ufw-user-input -p tcp -m tcp --dport 80 -j ACCEPT
> -A ufw-user-input -p udp -m udp --dport 80 -j ACCEPT
> -A ufw-user-limit -j REJECT --reject-with icmp-port-unreachable
> -A ufw-user-limit-accept -j ACCEPT
> -A ufw-user-logging-forward -j RETURN
> -A ufw-user-logging-input -j RETURN
> -A ufw-user-logging-output -j RETURN
> COMMIT
> # Completed on Fri Oct 14 11:04:15 2022

**Comment by tobiasbrunner** [comment, marked as answer]
> mark_in = %unique
> mark_out = %unique

Why did you configure this? If you do this, only packets with the assigned mark will be tunneled.

**Reply by Viktort-rf** [reply]
> > mark_in = %unique
> > mark_out = %unique
> 
> Why did you configure this? If you do this, only packets with the assigned mark will be tunneled.

If i have 2 client behind one NAT? This option may be help difference trafik or not?

**Reply by tobiasbrunner** [reply]
Not necessary when using IKEv2 with plain IPsec. That's only a problem when using IPsec in transport mode, which some L2TP clients do.

**Reply by Viktort-rf** [reply]
> Not necessary when using IKEv2 with plain IPsec. That's only a problem when using IPsec in transport mode, which some L2TP clients do.

e.g. if i have ikev2/l2tp-ipsec client, section _mark_in/out_ - desirable?

**Reply by tobiasbrunner** [reply]
I guess it depends on the client (some use random source ports or even tunnel mode). But I wouldn't recommend using L2TP anyway.

**Reply by Viktort-rf** [reply]
Thanks a lot for the replies and help!
p.s. delete mark_in/out help for ikev2-ipsec virtual-ip works
